geoextent/lib/helpfunctions.py

```diff
@@ -193,3 +193,114 @@ def date_parser(datetime_list, num_sample=None):
         parse_time = None
 
     return parse_time
+
+def extract_zip(zippedFile):
+    '''
+    Function purpose: unzip file (always inside a new folder)
+    Input: filepath
+    '''
+
+    abs_path = os.path.abspath(zippedFile)
+    root_folder = os.path.split(abs_path)[0]
+    zip_name = os.path.split(abs_path)[1][:-4]
+    zip_folder_path = os.path.join(root_folder, zip_name)
+
+    with zipfile.ZipFile(abs_path) as zipf:
+        zipf.extractall(zip_folder_path)
+
+def create_zip(folder):
+    '''
+    Function purpose: create a zip file
+    Input: filepath
+    Source: https://thispointer.com/python-how-to-create-a-zip-archive-from-multiple-files-or-directory/
+    '''
+
+    with zipfile.ZipFile(folder +'.zip', 'w') as zipObj:
+        # Iterate over all the files in directory
+        for folderName, subfolders, filenames in os.walk(folder):
+            for filename in filenames:
+                # create complete filepath of file in directory
+                file_root = os.path.abspath(folderName)
+                filePath = os.path.join(file_root, filename)
+                zipObj.write(filePath)
+
+
+def bbox_merge(metadata):
+    boxes_extent = []
+    for x, y in metadata.items():
+        if isinstance(y, dict):
+            try:
+                bbox_extent = [y['bbox'], y['crs']]
+                boxes_extent.append(bbox_extent)
+            except:
+                pass
+    if len(boxes_extent) == 0:
+        return None
+    elif len(boxes_extent) > 0:
+
+        multipolygon = ogr.Geometry(ogr.wkbMultiPolygon)
+        des_srs = ogr.osr.SpatialReference()
+        des_srs.ImportFromEPSG(WGS84_EPSG_ID)
+        multipolygon.AssignSpatialReference(des_srs)
+
+        for bbox in boxes_extent:
+
+            try:
+
+                box = ogr.Geometry(ogr.wkbLinearRing)
+                box.AddPoint(bbox[0][0], bbox[0][1])
+                box.AddPoint(bbox[0][2], bbox[0][1])
+                box.AddPoint(bbox[0][2], bbox[0][3])
+                box.AddPoint(bbox[0][0], bbox[0][3])
+                box.AddPoint(bbox[0][0], bbox[0][1])
+
+                if bbox[1] != "4326":
+                    logger.warning("Transforming bbox CRS from EPSG [{}] to [{}] ".format(bbox[1], WGS84_EPSG_ID))
+                    source = osr.SpatialReference()
+                    source.ImportFromEPSG(int(bbox[1]))
+                    transform = osr.CoordinateTransformation(source, des_srs)
+                    box.Transform(transform)
```

Comment:
Will this transformation never fail or throw an error?

Reply:
I'm using a "long" try except for this part. The most likely error in here is that the detected CRS name does not exist (Line 260). If there is an error creating the points or during the transformation, that geographical extent (CRS+BBOX) is ignored for the merging computation (no points are added to the 'merged' geometry). I'll add a debug message in the except statement to make it clear the specific error.

Reply:
Just to clarify: `.ImportFromEPSG` throws errors, but `.CoordinateTransformation` and `.Transform` do never throw errors but just return an empty geometry?